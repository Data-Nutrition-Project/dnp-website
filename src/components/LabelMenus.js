import React from "react"

const LabelMenus = props => {
  return (
    <div>
      <header></header>
      <span>
        <p></p>
      </span>
      <p></p>
    </div>
  )
}

export default LabelMenus
