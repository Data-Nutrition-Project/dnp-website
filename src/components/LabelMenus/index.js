import React, { Component } from "react"
<<<<<<< HEAD
import { connect } from "react-redux"
import { sendBaseInfo } from "../../store/bases"
class LabelMenus extends Component {
  constructor(props) {
    super(props)
    this.state = {
      showHighlight: this.props.highlightValue,
      menus: [
        {
          title: "Overview",
          desc:
            "Overview information about the dataset including Description, Provenance, Composition, and Collection.",
        },
        {
          title: "Use Cases and Alerts",
          desc:
            "Relevant alerts for data practitioners who intent to use this dataset for specific use cases (types of analyses).",
        },
        {
          title: "Maintenance",
          desc:
            "Information about the ongoing management of the dataset, such as how the data will be maintained, updated, and the best contact for further inquiries.",
        },
      ],
    }
    this.selectItem = this.selectItem.bind(this)
  }

  selectItem = item => {
    this.setState({
      selectedItem: item,
      showHighlight: false,
    })
    console.log("item", item)
    this.props.sendBaseInfo(item)
  }

  render() {
    return (
      <div
        className={styles.labelMenu}
        style={{ display: this.state.titleList }}
      >
        {this.state.menus.map((menu, id) => {
          return (
            <div
              id="selectButton"
              onClick={() => this.selectItem(menu.title)}
              className={
                this.state.selectedItem === menu.title
                  ? `${styles.highlightLabel} ${styles.labelMenus}`
                  : styles.labelMenus
              }
              key={id}
            >
              <div
                style={{
                  display: this.state.showHighlight ? "block" : "none",
                }}
              ></div>
              <span className={styles.menuTitle}>{menu.title}</span>
              <div className={styles.flexbox}>
                <span className={styles.weightedLine}></span>
              </div>
              <p className={styles.menuParagraph}>{menu.desc}</p>
            </div>
          )
        })}
      </div>
    )
  }
=======

import LabelTitle from "../LabelTitle"
import SectionBase from "../SectionBase"

import styles from "./styles.module.css"

class LabelMenus extends Component {
    constructor(props) {
        super(props)
        this.state = {
            showHighlight: this.props.highlightValue,
            titleList: ["Use Cases and Alerts", "Overview", "Maintenance"],
        }
    }

    selectItem = item =>
        this.setState({
            selectedItem: item,
            showHighlight: false,
        })
    render() {
        return (
            <div>
                <LabelTitle />
                <div className={styles.flexComponents}>
                    <div
                        className={styles.labelMenu}
                        style={{ display: this.state.titleList }}
                    >
                        {this.state.titleList.map((title, i) => {
                            return (
                                <div
                                    className={
                                        this.state.selectedItem === title
                                            ? `${styles.highlightLabel} ${styles.labelMenus}`
                                            : styles.labelMenus
                                    }
                                    id="selectButton"
                                    key={title.id}
                                    onClick={() => this.selectItem(title)}
                                    onKeyPress={() => this.selectItem(title)}
                                    role="button"
                                    tabIndex={i+1}
                                >
                                    <div
                                        key={title.id}
                                        style={{
                                            display: this.state.showHighlight ? "block" : "none",
                                        }}
                                    ></div>
                                    <span className={styles.menuTitle}>{title}</span>
                                    <div className={styles.flexbox}>
                                        <p className={styles.labelMenuDescription}>Lorem</p>
                                        <span className={styles.weightedLine}></span>
                                    </div>
                                    <p className={styles.menuParagraph}>
                                        Qui ut occaecat exercitation amet mollit excepteur do
                                        aliquip et sunt sit aliqua consectetur. Sit minim voluptate
                                        aliqua id do velit deserunt. Voluptate irure proident esse
                                        sint duis mollit culpa eiusmod officia ullamco aliqua in
                                        magna dolore.
                                    </p>
                                </div>
                            )
                        })}
                    </div>
                    <SectionBase />
                </div>
            </div>
        )
    }
>>>>>>> 4e75ff02b6d65beb9dc18c1f8e3acb7d875c6a62
}

const mapDispatch = dispatch => {
  return {
    sendBaseInfo: status => dispatch(sendBaseInfo(status)),
  }
}
export default connect(null, mapDispatch)(LabelMenus)
