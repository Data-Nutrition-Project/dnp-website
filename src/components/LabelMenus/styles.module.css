.labelMenu {
    width: 20%;
}

.labelMenus {
    height: 13rem;
    border: solid 0.0625rem #707070;
    background-color: #f4f4f4;
    margin: 2.1875rem 0rem;
    padding: 1.875rem 0rem 0rem 0rem;
}
.highlightLabel {
    background-image: url("/highlightbar.png");
    background-size: 0.5rem 35rem;
    background-repeat: no-repeat;
}

.labelMenuDescription {
    flex: 1;
    display: flex;
    width: 5.125rem;
    height: 0.875rem;
    font-family: Helvetica;
    font-size: 0.875rem;
    font-weight: normal;
    font-stretch: normal;
    font-style: normal;
    line-height: 1;
    letter-spacing: normal;
    text-align: left;
    margin: 0.25rem 0rem 0rem 1.46875rem;
}

.weightedLine {
    flex: 80% 0 0;
    border-bottom: 0.1875rem solid currentColor;
    display: inline;
    line-height: 0.85;
    margin: 0.46875rem 0rem 0rem 1.46875rem;
}

.menuParagraph {
    width: 15.375rem;
    height: 5.25rem;
    font-family: Helvetica;
    font-size: 0.75rem;
    font-weight: normal;
    font-stretch: normal;
    font-style: normal;
    line-height: 1;
    letter-spacing: normal;
    text-align: left;
    margin-left: 1.46875rem;
    margin-top: 0.53125rem;
}

.menuTitle {
    margin-left: 1.34375rem;
    width: 6.125rem;
    height: 1.375rem;
    font-family: Helvetica;
    font-size: 1.375rem;
    font-weight: bold;
    font-stretch: normal;
    font-style: normal;
    line-height: 1;
    letter-spacing: normal;
    text-align: left;
}

.flexbox {
    display: -webkit-flex;
    align-content: center;
    flex-wrap: wrap;
}

.flexComponents {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}
