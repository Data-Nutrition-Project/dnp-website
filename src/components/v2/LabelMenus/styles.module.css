@value layout: "../../layout.css";

@value tabletMaxWidth from layout;
@value landscapeMaxWidth from layout;
@value portraitMaxWidth from layout;

.menuTop {
  margin-bottom: 2rem;
}

.labelMenu {
  padding: 1.875rem 0 0 0;
  text-align: right;
  cursor: pointer;
  display: inline-block;
  position: relative;
  width: 100%;
}

.highlightMenuTitle {
  background-color: black;
  font-weight: bold;
  color: white;
  cursor: pointer;
}

.span {
  display: block;
  width: 100%;
}

.menuTitle {
  display: inline-block;
  margin-top: 0.5rem;
  font-size: 1.375rem;
  font-stretch: normal;
  font-style: normal;
  line-height: 1;
  letter-spacing: normal;
  text-align: right;
  padding: 0.5rem 2rem;
}

.menuTitle:hover {
  cursor: pointer;
}
.useButton .btn-primary {
  background-color: none !important;
}

@media (max-width: tabletMaxWidth) {
  .menuTitle {
    font-size: 0.8rem;
  }
}

@media (max-width: landscapeMaxWidth) {
  .menuTitle {
    font-size: 0.8rem;
  }

  .menuTop {
    margin-bottom: 1rem;
  }

  .labelMenu {
    display: flex;
    justify-content: space-around;
    padding: 0;
  }
}
