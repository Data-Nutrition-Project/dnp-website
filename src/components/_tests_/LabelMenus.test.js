import React from "react"
import renderer from "react-test-renderer"
import LabelMenus from "../LabelMenus"
import Adapter from "enzyme-adapter-react-16"
configure({ adapter: new Adapter() })
import { shallow, configure } from "enzyme"

describe("LabelMenus", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<LabelMenus />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test("selectItem should be called", () => {
    const props = {
      highlightValue: jest.fn(),
    }
    afterEach(() => {
      props.highlightValue.mockReset()
    })

    let wrapper = shallow(<LabelMenus {...props} />)
    wrapper.instance().selectItem = jest.fn()
    let { selectItem } = wrapper.instance()
    expect(selectItem).toHaveBeenCalledTimes(0)
    wrapper.find("#selectButton").forEach(node => {
      node.simulate("click")
    })
    expect(selectItem).toHaveBeenCalledTimes(3)
  })
})
