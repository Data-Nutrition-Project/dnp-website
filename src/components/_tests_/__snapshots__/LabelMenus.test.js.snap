// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelMenus renders correctly 1`] = `
<div>
  <div
    className="labelTitle"
  >
    <h1
      className="headerTitle"
    >
      Data Nutrition Label
    </h1>
    <div
      className="pageSubHeaderRow"
    >
      <h2
        className="datasetLabelHeader"
      >
        College Scoreboard Dataset 2020
      </h2>
      <a
        className="iconLink"
        href="#"
      >
        <img
          alt="icon for link"
          className="iconLinkImg"
          id="datasetLink"
          src="linkimg.png"
        />
      </a>
    </div>
    <div
      className="link"
    >
      <p
        className="datasetOriginLink"
      >
        US Department of Education
      </p>
    </div>
    <div
      className="link"
    >
      <a
        className="datasetLink"
        href="#"
      >
        <img
          alt=""
          id="datasetLink"
          src="/linkimg.png"
        />
      </a>
    </div>
    <span
      className="weightedLine"
    >
       
    </span>
  </div>
  <div
    className="flexComponents"
  >
    <div
      className="labelMenu"
      style={
        Object {
          "display": Array [
            "Use Cases and Alerts",
            "Overview",
            "Maintenance",
          ],
        }
      }
    >
      <div
        className="labelMenus"
        id="selectButton"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={1}
      >
        <div
          style={
            Object {
              "display": "none",
            }
          }
        />
        <span
          className="menuTitle"
        >
          Use Cases and Alerts
        </span>
        <div
          className="flexbox"
        >
          <p
            className="labelMenuDescription"
          >
            Lorem
          </p>
          <span
            className="weightedLine"
          />
        </div>
        <p
          className="menuParagraph"
        >
          Qui ut occaecat exercitation amet mollit excepteur do aliquip et sunt sit aliqua consectetur. Sit minim voluptate aliqua id do velit deserunt. Voluptate irure proident esse sint duis mollit culpa eiusmod officia ullamco aliqua in magna dolore.
        </p>
      </div>
      <div
        className="labelMenus"
        id="selectButton"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={2}
      >
        <div
          style={
            Object {
              "display": "none",
            }
          }
        />
        <span
          className="menuTitle"
        >
          Overview
        </span>
        <div
          className="flexbox"
        >
          <p
            className="labelMenuDescription"
          >
            Lorem
          </p>
          <span
            className="weightedLine"
          />
        </div>
        <p
          className="menuParagraph"
        >
          Qui ut occaecat exercitation amet mollit excepteur do aliquip et sunt sit aliqua consectetur. Sit minim voluptate aliqua id do velit deserunt. Voluptate irure proident esse sint duis mollit culpa eiusmod officia ullamco aliqua in magna dolore.
        </p>
      </div>
      <div
        className="labelMenus"
        id="selectButton"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={3}
      >
        <div
          style={
            Object {
              "display": "none",
            }
          }
        />
        <span
          className="menuTitle"
        >
          Maintenance
        </span>
        <div
          className="flexbox"
        >
          <p
            className="labelMenuDescription"
          >
            Lorem
          </p>
          <span
            className="weightedLine"
          />
        </div>
        <p
          className="menuParagraph"
        >
          Qui ut occaecat exercitation amet mollit excepteur do aliquip et sunt sit aliqua consectetur. Sit minim voluptate aliqua id do velit deserunt. Voluptate irure proident esse sint duis mollit culpa eiusmod officia ullamco aliqua in magna dolore.
        </p>
      </div>
    </div>
    <div
      className="sectionBase"
    >
      <h1
        className="sectionBaseHeader"
      >
         
      </h1>
      <span
        className="weightedLine"
      >
        <h2>
           
        </h2>
      </span>
      <span
        className="lightLine"
      >
        <h1>
           
        </h1>
      </span>
      <span
        className="lightLine"
      >
        <h1>
           
        </h1>
      </span>
    </div>
  </div>
</div>
`;
